addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -0,0 +1,190 @@
+        return f'{company.peppol_eas}:{company.peppol_endpoint}'
+
+    def _cron_peppol_get_new_documents(self):
+        # Retrieve all new PEPPOL documents for every edi user in the database
+        edi_users = self.env['account_edi_proxy_client.user'].search(
+            [('company_id.account_peppol_proxy_state', '=', 'active')])
+
+        proxy_acks = []
+        for edi_user in edi_users:
+            error_move_vals = {
+                'move_type': 'in_invoice',
+                'peppol_move_state': 'error',
+            }
+            try:
+                # request all messages that haven't been acknowledged
+                messages = edi_user._make_request(
+                    edi_user._get_server_url() + "/api/peppol/1/get_all_documents")
+            except AccountEdiProxyError as e:
+                messages = {'messages': []}
+                _logger.error(
+                    'Error while receiving the document from PEPPOL Proxy: %s', e.message)
+
+            message_uuids = [message['uuid']
+                             for message in messages['messages'] if message['direction'] == 'incoming']
+            if message_uuids:
+                company_id = edi_user.company_id
+                # retrieve attachments for filtered messages
+                all_messages = edi_user._make_request(
+                    edi_user._get_server_url() + "/api/peppol/1/get_document", dict(message_uuids=message_uuids))
+
+                for uuid, content in all_messages.items():
+                    if content.get('error'):
+                        # in this case there is no attachment that we could add to the account move
+                        move = self.env['account.move'].with_company(company_id).create(error_move_vals)
+                        move._message_log(body=content['error']['message'])
+                        continue
+
+                    enc_key = all_messages[uuid]["enc_key"]
+                    document_content = all_messages[uuid]["document"]
+                    decoded_document = edi_user._decrypt_data(document_content, enc_key)
+                    journal_id = company_id.peppol_purchase_journal_id
+                    # use the first purchase journal if the PEPPOL journal is not set up
+                    # to create the move anyway
+                    if not journal_id:
+                        journal_id = self.env['account.journal'].search(
+                            [('company_id', '=', company_id.id), ('type', '=', 'purchase')],
+                            limit=1
+                        )
+                    document_tree = etree.XML(decoded_document)
+                    filename = document_tree.find('{*}ID').text
+                    partner_endpoint = None
+                    if document_tree.find('{*}AccountingSupplierParty/{*}EndpointID'):
+                        partner_endpoint = document_tree.find('{*}AccountingSupplierParty/{*}EndpointID').text
```

Comment:
I thought it would be nice to get the proper filename and people wanted the partner endpoint to be logged in chatter. I am not sure how to get that information otherwise? :thinking: